builder/proxmox/clone/builder.go

```diff
@@ -51,10 +48,10 @@ type cloneVMCreator struct{}
 func (*cloneVMCreator) Create(vmRef *proxmoxapi.VmRef, config proxmoxapi.ConfigQemu, state multistep.StateBag) error {
 	client := state.Get("proxmoxClient").(*proxmoxapi.Client)
 	c := state.Get("clone-config").(*Config)
-	comm := state.Get("comm").(*communicator.Config)
+	comm := state.Get("config").(*proxmox.Config).Comm
```

Comment:
Nice improvement! But couldn't you do the same as you previously did with the ssh step above and just set it as a field on the `cloneVMCreator`?

Reply:
It's subtle, but we don't have the right references to pass it as a field there.

We've still got two layered builders, and two separate config references - one is the common builder+config, the other is the clone (or iso) config. The values of each are the same, but they are not the same reference. The ssh setup needs to modify a `config.Comm` in order to work, and since the common task is the one that connects, it is the common builder's config that needs to be edited.

As far as I understand, the reference to the common builder's config is still a different one than when we're in the clone builder, when using `NewSharedBuilder`.

I was adding the Comm ref into the statebag, and using that on connect before. I've now swapped it to be using the common config's Comm ref, but for how things are set up currently, these two are still not the same object.

I agree that it'd be better if there's a way of unifying the config so that it's the same config object for both builder references. Then we could make it nice and pass the clone config's Comm in, and it'd be the same reference as common's Comm. I'm just not sure 100% how to update the NewSharedBuilder so that the config ref is the same.

Reply:
Okay, I'm fine with leaving this as a battle for another day :)